src/LockGuard.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace Mesavolt\FiveOhThree;
+
+
+abstract class LockGuard
+{
+    private const DEFAULTS = [
+        'lock_path' => __DIR__.'/../../../../deploying.lock',
+        'template' => __DIR__.'/../res/template.php',
+
+        // default template configuration
+        'auto_refresh' => true,
+        'auto_refresh_interval' => 8,
+        'icon' => '/apple-touch-icon.png',
+    ];
+
+    /**
+     * Checks if a lock file exists.
+     * Set the FIVEOHTHREE_IGNORE_LOCK environment variable to a true-ish value (decoded with `boolval`)
+     * to ignore the presence of a lock file and always return `false` (useful in deployment scripts).
+     */
+    public static function lockFileExists(string $path): bool
+    {
+        $ignoreFlag = boolval($_ENV['FIVEOHTHREE_IGNORE_LOCK'] ?? false);
```

Comment:
What about using a cast instead of `boolval`? There's a slight overhead in using a function call - since this code is executed on each request, we should optimize it a bit, what do you think?

```suggestion
        $ignoreFlag = (bool) ($_ENV['FIVEOHTHREE_IGNORE_LOCK'] ?? false);
```